platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -64,47 +68,157 @@
+        collection,
+        zookeeperHosts);
+
+    String alias = properties == null ? null : (String) properties.getOrDefault(ALIAS_PROP, null);
+
     return new SolrClientAdapter(
-        core,
+        collection,
         () ->
             AccessController.doPrivileged(
                 (PrivilegedAction<SolrClient>)
                     () -> {
-                      return createSolrCloudClient(zookeeperHosts, core);
+                      return createSolrCloudClient(zookeeperHosts, collection, alias);
                     }));
   }
 
   @VisibleForTesting
-  SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+  void addCollectionToAlias(String alias, String collection, CloudSolrClient client) {
+    if (StringUtils.isBlank(alias) || StringUtils.isBlank(collection)) {
+      return;
+    }
+    try {
+      List<String> aliasedCollections = getCollectionsInAlias(client, alias);
+      if (!aliasedCollections.contains(collection)) {
+        List<String> collections = new ArrayList<>();
+        collections.addAll(aliasedCollections);
+        collections.add(collection);
+        collections.addAll(getCollectionsNotInAlias(client, aliasedCollections, alias));
+        createAlias(client, collections, alias);
+      }
+    } catch (SolrServerException | IOException e) {
+      LOGGER.warn("Failed to update alias [{}]", alias, e);
+    }
+  }
+
+  private List<String> getCollectionsInAlias(CloudSolrClient client, String alias)
+      throws SolrServerException, IOException {
+    CollectionAdminResponse aliasResponse =
+        new CollectionAdminRequest.ListAliases().process(client);
+
+    if (aliasResponse != null) {
+      Map<String, String> aliases = aliasResponse.getAliases();
+      if (aliases != null && aliases.containsKey(alias)) {
+        String[] collections = aliases.get(alias).split(",");
+        return Arrays.asList(collections);
+      }
+    }
+    return Collections.emptyList();
+  }
+
+  /**
+   * Find existing collections in case parallel operations are creating collections
+   *
+   * @param client
+   * @param currentAliases
+   * @param alias
+   * @return
+   * @throws SolrServerException
+   * @throws IOException
+   */
+  private List<String> getCollectionsNotInAlias(
+      CloudSolrClient client, List<String> currentAliases, String alias)
+      throws SolrServerException, IOException {
+    if (StringUtils.isNotBlank(alias)) {
+      CollectionAdminResponse response = new CollectionAdminRequest.List().process(client);
+      if (response.getResponse() != null) {
+        List<String> collections = (List<String>) response.getResponse().get("collections");
+        return collections == null
+            ? Collections.emptyList()
+            : collections
+                .stream()
+                .filter(c -> c.startsWith(alias))
+                .filter(c -> currentAliases.contains(c))
+                .collect(Collectors.toList());
```

Comment:
✏️ Bummer optionals and exceptions dont play well together or you could probably do more with this. But even the latter half could be Optionaled if you'd prefer
```java
return Optional.ofNullable(response)
    .map(CollectionAdminResponse::getResponse)
    .map(resp -> resp.get("collections"))
    .orElse(Collections.emptyList())
    .stream()
    .filter(c -> c.startsWith(alias))
    .filter(c -> currentAliases.contains(c))
    .collect(Collectors.toList());
```